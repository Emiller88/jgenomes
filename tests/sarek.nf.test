nextflow_pipeline {

    name "Build sarek references"
    script "../main.nf"
    tag "pipeline"

    test("-profile test --tools bwamem1,bwamem2,createsequencedictionary,dragmap,faidx,intervals,msisensorpro,tabix --input references/assets GRCh38_chr22_source.yml") {

        options "-output-dir $outputDir"

        when {
            params {
                input  = "https://raw.githubusercontent.com/nf-core/references-assets/main/genomes_source/Homo_sapiens/test/GRCh38_chr22.yml"
                outdir = "$outputDir"
                tools  = 'bwamem1,bwamem2,createsequencedictionary,dragmap,faidx,intervals,msisensorpro,tabix'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_references_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }

    test("-profile test --tools bwamem1,bwamem2,createsequencedictionary,dragmap,faidx,intervals,msisensorpro,tabix --input references/assets GRCh38_chr22_published.yml") {

        options "-output-dir $outputDir"

        when {
            params {
                input  = "https://raw.githubusercontent.com/nf-core/references-assets/main/genomes_no_base/Homo_sapiens/test/GRCh38_chr22.yml"
                outdir = "$outputDir"
                tools  = 'bwamem1,bwamem2,createsequencedictionary,dragmap,faidx,intervals,msisensorpro,tabix'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_references_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
