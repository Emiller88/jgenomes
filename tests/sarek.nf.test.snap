{
    "Run with profile test for sarek | --input assets/test/pipelines/sarek.yml": {
        "content": [
            9,
            {
                "BUILD_INTERVALS": {
                    "gawk": "5.3.0"
                },
                "BWAMEM1_INDEX": {
                    "bwa": "0.7.18-r1243-dirty"
                },
                "BWAMEM2_INDEX": {
                    "bwamem2": "2.2.1"
                },
                "DRAGMAP_HASHTABLE": {
                    "dragmap": "1.2.1"
                },
                "SAMTOOLS_FAIDX": {
                    "samtools": 1.21
                },
                "TABIX_TABIX": {
                    "tabix": 1.2
                },
                "Workflow": {
                    "nf-core/references": "v1.0.0dev"
                }
            },
            [
                "Homo_sapiens",
                "Homo_sapiens/nf-core",
                "Homo_sapiens/nf-core/references",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Annotation",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Annotation/GATK_BUNDLE",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Annotation/GATK_BUNDLE/dbsnp_146.hg38.vcf.gz.tbi",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Annotation/GATK_BUNDLE/gnomAD.r2.1.1.vcf.gz.tbi",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Annotation/GATK_BUNDLE/mills_and_1000G.indels.vcf.gz.tbi",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Annotation/intervals",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Annotation/intervals/testdata.GRCh38_chr22.bed",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAIndex",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAIndex/version0.7.18",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAIndex/version0.7.18/genome.amb",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAIndex/version0.7.18/genome.ann",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAIndex/version0.7.18/genome.bwt",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAIndex/version0.7.18/genome.pac",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAIndex/version0.7.18/genome.sa",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAmem2Index",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAmem2Index/version2.2.1",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAmem2Index/version2.2.1/genome.fasta.0123",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAmem2Index/version2.2.1/genome.fasta.amb",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAmem2Index/version2.2.1/genome.fasta.ann",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAmem2Index/version2.2.1/genome.fasta.bwt.2bit.64",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAmem2Index/version2.2.1/genome.fasta.pac",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/WholeGenomeFasta",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/WholeGenomeFasta/genome.fasta.fai",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/dragmap",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/dragmap/version1.2.1",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/dragmap/version1.2.1/hash_table.cfg",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/dragmap/version1.2.1/hash_table.cfg.bin",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/dragmap/version1.2.1/hash_table.cmp",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/dragmap/version1.2.1/hash_table_stats.txt",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/dragmap/version1.2.1/ref_index.bin",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/dragmap/version1.2.1/reference.bin",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/dragmap/version1.2.1/repeat_mask.bin",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/dragmap/version1.2.1/str_table.bin",
                "index.json",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_references_software_mqc_versions.yml"
            ],
            [
                "dbsnp_146.hg38.vcf.gz.tbi:md5,628232d0c870f2dbf73c3e81aff7b4b4",
                "gnomAD.r2.1.1.vcf.gz.tbi:md5,d95bc4bd1eee441d13ed99f0f3d501f8",
                "mills_and_1000G.indels.vcf.gz.tbi:md5,1bb7ab8f22eb798efd796439d3b29b7a",
                "testdata.GRCh38_chr22.bed:md5,87a15eb9c2ff20ccd5cd8735a28708f7",
                "genome.amb:md5,1891c1de381b3a96d4e72f590fde21.0.0dev",
                "genome.ann:md5,2df4aa2d7580639fa0fcdbcad5e2e969",
                "genome.bwt:md5,815eded87e4cb6b1.0.0devdaab5c4d6e30af",
                "genome.pac:md5,8569fbdb2c98c6fb16dfa73d8eacb070",
                "genome.sa:md5,e7cff62b919448a3a3d0fe4aaf427594",
                "genome.fasta.0123:md5,d73300d44f733bcdb7c988fc3ff3e3e9",
                "genome.fasta.amb:md5,1891c1de381b3a96d4e72f590fde21.0.0dev",
                "genome.fasta.ann:md5,2df4aa2d7580639fa0fcdbcad5e2e969",
                "genome.fasta.bwt.2bit.64:md5,cd4bdf496eab05228a50c45ee43c1ed0",
                "genome.fasta.pac:md5,8569fbdb2c98c6fb16dfa73d8eacb070",
                "genome.fasta.fai:md5,3520cd31.0.0devb100e55f578db9c855f685",
                "hash_table.cmp:md5,1caab4ffc89f81ace615a2e813295cf4",
                "ref_index.bin:md5,dbb5c7d26b974e0ac338024fe4535044",
                "reference.bin:md5,be67b80ee48aa96b383fd72f1ccfefea",
                "repeat_mask.bin:md5,294939f1f80aa7f4a70b9b537e4c0f21",
                "str_table.bin:md5,45f7818c4a10fdeed04db7a34b5f9ff1",
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-11.0.0dev3:45:37.140557415"
    }
}
